src/managers/queue-manager.js

```diff
@@ -78,30 +78,36 @@ async function handleMessage(message) {
   }
 }
 
-export async function pollMessageQueue() {
+export async function pullMessagesFromQueue() {
   if (getUserToken()) {
     if (Gist.isDocumentVisible) {
-      var response = await getUserQueue();
-      var responseData = [];
-      if (response) {
-        if (response.status === 200 || response.status === 204) {
-          var expiryDate = new Date(new Date().getTime() + 1 * 60000);
-          setKeyWithExpiryToLocalStore(userQueueLocalStoreName, response.data, expiryDate);
-          responseData = response.data;
-        }
-        else if (response.status === 304) {
-          responseData = getKeyFromLocalStore(userQueueLocalStoreName);
-        }
-        if (responseData && responseData.length > 0) {
-          log(`Message queue checked for user ${getUserToken()}, ${responseData.length} messages found.`);
-          messages = responseData;
-          await checkMessageQueue();
+      // We're using the TTL as a way to determine if we should check the queue, so if the key is not there, we check the queue.
+      if (getKeyFromLocalStore(userQueueNextPullCheckLocalStoreName) === null) {
+        var response = await getUserQueue();
+        var responseData = [];
+        if (response) {
+          if (response.status === 200 || response.status === 204) {
+            var expiryDate = new Date(new Date().getTime() + 1 * 60000);
```

Comment:
Maybe the 60000 can be written as a constant, too?

Reply:
Good call, will also increase the local store cache to 1 hour instead of 1min (to reflect recent backend changes)